src/components/theme/Header/index.jsx

```diff
@@ -1,16 +1,26 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
+import { ThemeContext } from 'providers/ThemeProvider';
 import Navbar from './Navbar';
 import Hamburger from './Hamburger';
 import Sidebar from './Sidebar';
 import { Wrapper, Overlay } from './styles';
 
 export const Header = () => {
   const [sidebar, toggle] = useState(false);
+  const [hasScrolled, setNavBar] = useState(false);
+
+  if (typeof document !== `undefined`) {
+    document.addEventListener('scroll', () => {
+      const threshold = 80;
+      let scrolledAboveThreshold = window.scrollY >= threshold ? true : false;
+      setNavBar(scrolledAboveThreshold);
+    });
```

Comment:
Hi @ankitbeniwal ! I just want to thank you again for these changes. I made some of my own modifications from the last PR and I thought this might be a good opportunity share some additional information that I found in case this might help you later :) 

### 1. "document is not defined" error
I moved the original code into header component because we don't need to listen for events anywhere else. I then wrapped the function with a check against `document`. Actually, it turns out that in Gatsby it uses nodejs to run a local server. But when we run `gatsby build` it creates a build error like below

<details>
  <summary>See error log here</summary>

```bash
failed Building static HTML for pages - 77.967s

 ERROR #95312

"document" is not available during server side rendering.

See our docs page for more info on this error: https://gatsby.dev/debug-html


   9 |   const [navBar, setNavBar] = useState(false);
  10 |
> 11 |   document.addEventListener('scroll', () => {
     |   ^
  12 |     const threshold = 80;
  13 |     let scrolledAboveThreshold = window.scrollY >= threshold ? true : false;
  14 |     setNavBar(scrolledAboveThreshold);


  WebpackError: ReferenceError: document is not defined

  - index.jsx:11 Header
    src/components/theme/Header/index.jsx:11:3
```
</details>

Ref: https://www.gatsbyjs.com/docs/troubleshooting-common-errors/#error-referenceerror-window-is-not-defined-when-running-gatsby-build

### 2. Removed `setInterval`

Actually I think this can be removed because this function is used to trigger some action after xxx miliseconds. Since we added an event listener it already checks for events by the user whenever that action is being done.

Reply:
@benji011 I am happy to help and contribute. 

As far as the ```setInterval``` is concerned, I used the [Throttling Design Pattern](https://codeburst.io/throttling-and-debouncing-in-javascript-b01cad5c8edf) in order to boost the performance a little. Anyways, you may decide which method you want as per your requirements. 

Thanks for sharing this information and letting me contribute.

Reply:
@benji011 I am happy to help and contribute. 

As far as the ```setInterval``` is concerned, I used the [Throttling Design Pattern](https://codeburst.io/throttling-and-debouncing-in-javascript-b01cad5c8edf) in order to boost the performance a little. Anyway, you may decide which method you want as per your requirements. 

Thanks for sharing this information and letting me contribute.